### PR TITLE
EOS-20365: Health on CLI

### PR DESCRIPTION
**Pull request by rohitkolapkar**
# Backend

# Problem Statement

[EOS-20365: CLI: Node Health](https://jts.seagate.com/browse/EOS-20365)
- Add new command in cortxcli that will return the health of the cluster and the resources in it.

## Design

- _For Bug describe the fix here._
- _For feature, post the link to the solution page._
[Health CLI Specification](https://seagate-systems.atlassian.net/wiki/spaces/PRIVATECOR/pages/380141918/CLI+Specification)
- _Any northbound interface change and has been notified to other gatekeepers [Y/N]:_ NA

## Coding

- _Coding conventions are followed and code is consistent. Yes/No?_ Yes
- _Confirm All CODACY errors are resolved. Yes/No?_ Yes

## Testing

- _Confirm that Test Cases are added (for both the cases, fix and feature). Yes/No?_ Yes
- _Confirm Test Cases cover Happy Path, Non-Happy Path and Scalability. Yes/No?_ Yes
- _Confirm Testing was performed with installed RPM. Yes/No?_  No
- _Confirm all the ut/regression/smoke test has been performed [Y/N]_ No

## Checklist

- _PR is self-reviewed? Yes/No?_ Yes
- _GitHub Issue is updated_
- _Jira is updated_
  -  _Check if the description is clear and explained._ 
    -  _Check Acceptance Criterion is defined._
      -  _All the tests performed should be mentioned before Resolving a JIRA._
        -  _Verification needs to be done before marked as Closed/Verified_
        - _Any interface change Yes/No?_ No
        - _Change notified to other gatekeepers: Yes/No?_ NA
        - _Code reviews done and addressed: Yes/No?_ Yes
        - _Codacy issues reviewed and addressed: Yes/No?_ Yes
        - _Deployment test : Done/Not?_ Yes
        - _UT/smoke test: Done/Not?_ 
        - _Jira number added to PR: Yes/No?_ Yes
        - _Github merge done using UI: Yes/No?_
        - _Side effects on other features - deployment/update/node-replacement_
        - _Changes to quick-start-guide/community impact_
        - _All dependent component code PR are raised or already merged Yes/No_
        - _All dependent code already merged in landing branch Yes/No_ 
        
Signed-off-by: rohitkolapkar <rohit.j.kolapkar@seagate.com>

### CODE REVIEW

File: csm/cli/schema/health.json [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+      {
+        "name": "show",
+        "description": "Displays health of the cluster and resources in it.",
+        "need_confirmation": false,
+        "permissions_tag": "list",
+        "args": [
+          {
+            "flag": "resource",
+            "type": "str",
+            "help": "The type of element in the hierarchy. Ex. node"
```
Comment (udayan-y21) [inline csm/cli/schema/health.json:14]: Is there a list of choices for `resource` ? What are the possible values for these argumant?

File: csm/cli/schema/health.json [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            "params": true
+          },
+          {
+            "flag": "-r",
+            "dest": "response_format",
+            "type": "str",
+            "default": "table",
+            "suppress_help": true,
+            "help": "",
+            "params": true
```
Comment (udayan-y21) [inline csm/cli/schema/health.json:47]: How is different from `-f` flag? `help` is suppressed ..user wont able to find difference between `-r` and `-f` flags

**Review by udayan-y21 — CHANGES_REQUESTED** [review]

File: csm/cli/schema/health.json (hunk 1 above)
Comment (rohitkolapkar) [reply csm/cli/schema/health.json:14]: @udayan-y21 
yes, there is a list of choices. for eg. [cluster, site, rack , node] and in future that list will grow. 
And as discussed with @pawankumarsrivastava, possible list of arguments will be mentioned in CLI command specification documents.

File: csm/cli/schema/health.json (hunk 2 above)
Comment (rohitkolapkar) [reply csm/cli/schema/health.json:47]: @udayan-y21, 
REST api is strictly expecting a query param : response_format=table. This param will return response data in single array which is required for tabular format view. And user is not allowed to change the default value of this param that is why suppress_help is set to true.

**Review by soniyamoholkar — APPROVED** [review]

**Pull request closed without merging**